<a href="https://colab.research.google.com/github/Chabbbbbo/EXPLORATION/blob/main/EXPLORATION_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**1-1 Digit**

In [1]:
#데이터 불러오기
from sklearn.datasets import load_digits
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report

#데이터 준비 및 정보확인
digits= load_digits()
print(dir(digits))

#데이터 이해하기 
digits_data = digits.data #Feature Data 저장
digits_label = digits.target #Label Data 저장
print(digits.target_names) #Target Names 출력
print(digits.DESCR) #데이터 describe 해보기

#Train, test 데이터 분리
X_train, X_test, y_train, y_test = train_test_split(digits_data, digits_label, test_size=0.2)




['DESCR', 'data', 'feature_names', 'frame', 'images', 'target', 'target_names']
[0 1 2 3 4 5 6 7 8 9]
.. _digits_dataset:

Optical recognition of handwritten digits dataset
--------------------------------------------------

**Data Set Characteristics:**

    :Number of Instances: 1797
    :Number of Attributes: 64
    :Attribute Information: 8x8 image of integer pixels in the range 0..16.
    :Missing Attribute Values: None
    :Creator: E. Alpaydin (alpaydin '@' boun.edu.tr)
    :Date: July; 1998

This is a copy of the test set of the UCI ML hand-written digits datasets
https://archive.ics.uci.edu/ml/datasets/Optical+Recognition+of+Handwritten+Digits

The data set contains images of hand-written digits: 10 classes where
each class refers to a digit.

Preprocessing programs made available by NIST were used to extract
normalized bitmaps of handwritten digits from a preprinted form. From a
total of 43 people, 30 contributed to the training set and different 13
to the test set. 32x32 bit

In [2]:
#1) Decision Tree
from sklearn.tree import DecisionTreeClassifier
decision_tree = DecisionTreeClassifier(random_state=32)
decision_tree.fit(X_train, y_train)

y_pred = decision_tree.predict(X_test)

print(classification_report(y_test, y_pred))


              precision    recall  f1-score   support

           0       0.88      0.88      0.88        26
           1       0.81      0.88      0.84        33
           2       0.90      0.70      0.79        40
           3       0.88      0.88      0.88        43
           4       0.85      0.80      0.82        35
           5       0.95      0.87      0.91        45
           6       0.95      0.91      0.93        45
           7       0.88      0.97      0.93        38
           8       0.62      0.89      0.73        27
           9       0.85      0.79      0.81        28

    accuracy                           0.86       360
   macro avg       0.86      0.86      0.85       360
weighted avg       0.87      0.86      0.86       360



In [3]:
#2) Random Forest
from sklearn.ensemble import RandomForestClassifier

randomforest = RandomForestClassifier(random_state=32)
randomforest.fit(X_train, y_train)

y_pred = randomforest.predict(X_test)

print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       0.93      1.00      0.96        26
           1       1.00      1.00      1.00        33
           2       1.00      0.97      0.99        40
           3       1.00      0.98      0.99        43
           4       1.00      0.91      0.96        35
           5       0.96      0.98      0.97        45
           6       1.00      0.98      0.99        45
           7       0.93      1.00      0.96        38
           8       0.87      0.96      0.91        27
           9       1.00      0.89      0.94        28

    accuracy                           0.97       360
   macro avg       0.97      0.97      0.97       360
weighted avg       0.97      0.97      0.97       360



In [4]:
#3) SVM
from sklearn import svm
svm_model = svm.SVC()

svm_model.fit(X_train, y_train)
y_pred = svm_model.predict(X_test)

print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       1.00      1.00      1.00        26
           1       1.00      1.00      1.00        33
           2       1.00      1.00      1.00        40
           3       0.98      0.98      0.98        43
           4       1.00      0.94      0.97        35
           5       0.96      1.00      0.98        45
           6       1.00      1.00      1.00        45
           7       0.97      1.00      0.99        38
           8       0.96      1.00      0.98        27
           9       0.96      0.89      0.93        28

    accuracy                           0.98       360
   macro avg       0.98      0.98      0.98       360
weighted avg       0.98      0.98      0.98       360



In [5]:
# 4)SGD Classifier
from sklearn.linear_model import SGDClassifier
sgd_model = SGDClassifier()

sgd_model.fit(X_train, y_train)
y_pred = sgd_model.predict(X_test)

print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       1.00      1.00      1.00        26
           1       0.87      1.00      0.93        33
           2       1.00      1.00      1.00        40
           3       0.98      0.93      0.95        43
           4       1.00      0.94      0.97        35
           5       0.90      0.98      0.94        45
           6       1.00      0.93      0.97        45
           7       0.97      1.00      0.99        38
           8       0.93      0.93      0.93        27
           9       1.00      0.89      0.94        28

    accuracy                           0.96       360
   macro avg       0.96      0.96      0.96       360
weighted avg       0.96      0.96      0.96       360



In [6]:
# 5)Logistic Regression
from sklearn.linear_model import LogisticRegression
logistic_model = LogisticRegression(max_iter=5000)

logistic_model.fit(X_train, y_train)
y_pred = logistic_model.predict(X_test)

print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       1.00      1.00      1.00        26
           1       0.89      1.00      0.94        33
           2       1.00      1.00      1.00        40
           3       1.00      0.95      0.98        43
           4       0.97      0.91      0.94        35
           5       0.91      0.96      0.93        45
           6       1.00      0.96      0.98        45
           7       0.95      1.00      0.97        38
           8       0.93      0.96      0.95        27
           9       1.00      0.89      0.94        28

    accuracy                           0.96       360
   macro avg       0.97      0.96      0.96       360
weighted avg       0.97      0.96      0.96       360



1-1의 모델 평가  

해당 분류는 정확도가 0.98로 가장 높은 SVM을 사용해서 평가하는것이 좋다고 생각됨




**1-2 wine**

In [7]:
from sklearn.datasets import load_wine
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report

#데이터 준비 및 정보확인
wine= load_wine()
print(dir(wine))

#데이터 이해하기 
wine_data = wine.data #Feature Data 저장
wine_label = wine.target #Label Data 저장
print(wine.target_names) #Target Names 출력
print(wine.DESCR) #데이터 describe 해보기

#Train, test 데이터 분리
X_train, X_test, y_train, y_test = train_test_split(wine_data, wine_label, test_size=0.2, random_state=5)


['DESCR', 'data', 'feature_names', 'frame', 'target', 'target_names']
['class_0' 'class_1' 'class_2']
.. _wine_dataset:

Wine recognition dataset
------------------------

**Data Set Characteristics:**

    :Number of Instances: 178 (50 in each of three classes)
    :Number of Attributes: 13 numeric, predictive attributes and the class
    :Attribute Information:
 		- Alcohol
 		- Malic acid
 		- Ash
		- Alcalinity of ash  
 		- Magnesium
		- Total phenols
 		- Flavanoids
 		- Nonflavanoid phenols
 		- Proanthocyanins
		- Color intensity
 		- Hue
 		- OD280/OD315 of diluted wines
 		- Proline

    - class:
            - class_0
            - class_1
            - class_2
		
    :Summary Statistics:
    
    ============================= ==== ===== ======= =====
                                   Min   Max   Mean     SD
    ============================= ==== ===== ======= =====
    Alcohol:                      11.0  14.8    13.0   0.8
    Malic Acid:                   0.74  5.80    2.3

In [8]:
#1) Decision Tree
from sklearn.tree import DecisionTreeClassifier
decision_tree = DecisionTreeClassifier(random_state=32)
decision_tree.fit(X_train, y_train)

y_pred = decision_tree.predict(X_test)

print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       1.00      0.76      0.87        17
           1       0.73      1.00      0.85        11
           2       0.88      0.88      0.88         8

    accuracy                           0.86        36
   macro avg       0.87      0.88      0.86        36
weighted avg       0.89      0.86      0.86        36



In [9]:
#2) Random Forest
from sklearn.ensemble import RandomForestClassifier

randomforest = RandomForestClassifier(random_state=32)
randomforest.fit(X_train, y_train)

y_pred = randomforest.predict(X_test)

print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       1.00      1.00      1.00        17
           1       1.00      1.00      1.00        11
           2       1.00      1.00      1.00         8

    accuracy                           1.00        36
   macro avg       1.00      1.00      1.00        36
weighted avg       1.00      1.00      1.00        36



In [10]:
#3) SVM
from sklearn import svm
svm_model = svm.SVC()

svm_model.fit(X_train, y_train)
y_pred = svm_model.predict(X_test)

print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       0.86      0.71      0.77        17
           1       0.69      0.82      0.75        11
           2       0.44      0.50      0.47         8

    accuracy                           0.69        36
   macro avg       0.66      0.67      0.66        36
weighted avg       0.72      0.69      0.70        36



In [11]:
# 4)SGD Classifier
from sklearn.linear_model import SGDClassifier
sgd_model = SGDClassifier()

sgd_model.fit(X_train, y_train)
y_pred = sgd_model.predict(X_test)

print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       0.81      0.76      0.79        17
           1       0.50      0.91      0.65        11
           2       0.00      0.00      0.00         8

    accuracy                           0.64        36
   macro avg       0.44      0.56      0.48        36
weighted avg       0.54      0.64      0.57        36



/usr/local/lib/python3.7/dist-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
/usr/local/lib/python3.7/dist-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))
/usr/local/lib/python3.7/dist-packages/sklearn/metrics/_classification.py:1318: UndefinedMetricWarning: Precision and F-score are ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, msg_start, len(result))


In [12]:
# 5)Logistic Regression
from sklearn.linear_model import LogisticRegression
logistic_model = LogisticRegression(max_iter=5000)

logistic_model.fit(X_train, y_train)
y_pred = logistic_model.predict(X_test)

print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       1.00      0.94      0.97        17
           1       0.85      1.00      0.92        11
           2       1.00      0.88      0.93         8

    accuracy                           0.94        36
   macro avg       0.95      0.94      0.94        36
weighted avg       0.95      0.94      0.95        36



1-2 모델평가  
해당 분류는 accuracy가 1.0로 가장 높은 RandomForest를 사용하여 평가하는 것이 좋다고 생각됨 

**1-3 load_breast_cancer**

In [13]:
from sklearn.datasets import load_breast_cancer
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report

#데이터 준비 및 정보확인
cancer= load_breast_cancer()
print(dir(cancer))

#데이터 이해하기 
cancer_data = cancer.data #Feature Data 저장
cancer_label = cancer.target #Label Data 저장
print(cancer.target_names) #Target Names 출력
print(cancer.DESCR) #데이터 describe 해보기

#Train, test 데이터 분리
X_train, X_test, y_train, y_test = train_test_split(cancer_data, cancer_label, test_size=0.2, random_state=5)


['DESCR', 'data', 'data_module', 'feature_names', 'filename', 'frame', 'target', 'target_names']
['malignant' 'benign']
.. _breast_cancer_dataset:

Breast cancer wisconsin (diagnostic) dataset
--------------------------------------------

**Data Set Characteristics:**

    :Number of Instances: 569

    :Number of Attributes: 30 numeric, predictive attributes and the class

    :Attribute Information:
        - radius (mean of distances from center to points on the perimeter)
        - texture (standard deviation of gray-scale values)
        - perimeter
        - area
        - smoothness (local variation in radius lengths)
        - compactness (perimeter^2 / area - 1.0)
        - concavity (severity of concave portions of the contour)
        - concave points (number of concave portions of the contour)
        - symmetry
        - fractal dimension ("coastline approximation" - 1)

        The mean, standard error, and "worst" or largest (mean of the three
        worst/largest value

In [14]:
#1) Decision Tree
from sklearn.tree import DecisionTreeClassifier
decision_tree = DecisionTreeClassifier(random_state=32)
decision_tree.fit(X_train, y_train)

y_pred = decision_tree.predict(X_test)

print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       0.94      0.92      0.93        48
           1       0.94      0.95      0.95        66

    accuracy                           0.94       114
   macro avg       0.94      0.94      0.94       114
weighted avg       0.94      0.94      0.94       114



In [15]:
#2) Random Forest
from sklearn.ensemble import RandomForestClassifier

randomforest = RandomForestClassifier(random_state=32)
randomforest.fit(X_train, y_train)

y_pred = randomforest.predict(X_test)

print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       1.00      0.94      0.97        48
           1       0.96      1.00      0.98        66

    accuracy                           0.97       114
   macro avg       0.98      0.97      0.97       114
weighted avg       0.97      0.97      0.97       114



In [16]:
#3) SVM
from sklearn import svm
svm_model = svm.SVC()

svm_model.fit(X_train, y_train)
y_pred = svm_model.predict(X_test)

print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       1.00      0.85      0.92        48
           1       0.90      1.00      0.95        66

    accuracy                           0.94       114
   macro avg       0.95      0.93      0.94       114
weighted avg       0.94      0.94      0.94       114



In [17]:
# 4)SGD Classifier
from sklearn.linear_model import SGDClassifier
sgd_model = SGDClassifier()

sgd_model.fit(X_train, y_train)
y_pred = sgd_model.predict(X_test)

print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       0.70      0.98      0.82        48
           1       0.98      0.70      0.81        66

    accuracy                           0.82       114
   macro avg       0.84      0.84      0.82       114
weighted avg       0.86      0.82      0.82       114



In [19]:
# 5)Logistic Regression
from sklearn.linear_model import LogisticRegression
logistic_model = LogisticRegression(max_iter=5000)

logistic_model.fit(X_train, y_train)
y_pred = logistic_model.predict(X_test)

print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       0.98      0.94      0.96        48
           1       0.96      0.98      0.97        66

    accuracy                           0.96       114
   macro avg       0.97      0.96      0.96       114
weighted avg       0.97      0.96      0.96       114



1-2 모델평가  
해당 분류는 accuracy가 0.97로 가장 높은 RandomForest 기법을 사용하는 것이 가장 좋다고 생각됨